pipe-cli/src/utilities/dts_operations_manager.py

```diff
@@ -0,0 +1,99 @@
+
+
+class DtsOperationsManager:
+
+    _DTS_TABLE_HEADERS = ['Registry ID', 'Name', 'URL', 'Created date', 'Schedulable', 'Prefixes', 'Preferences']
+    _PREF_DELIMITER = "="
+
+    def __init__(self):
+        pass
+
+    def create(self, url, name, schedulable, prefixes, preferences, json_out):
+        registry = DTS.create(url, name, schedulable, prefixes, self._convert_preferences_list_to_dict(preferences))
+        self._print_registry(registry, json_out)
+
+    def list(self, registry_id, json_out):
+        if registry_id:
+            registry = DTS.load(registry_id)
+            self._print_registry(registry, json_out)
+        else:
+            registries = DTS.load_all()
+            if json_out:
+                self._print_dts_json(registries)
+            else:
+                self._print_registries_prettytable(registries)
+
+    def upsert_preferences(self, registry_id, preferences_list, json_out):
+        if not preferences_list:
+            click.echo('Preferences should not be empty!', err=True)
+            sys.exit(1)
+        updated_registry = DTS.update_preferences(registry_id, self._convert_preferences_list_to_dict(preferences_list))
+        self._print_registry(updated_registry, json_out)
+
+    def delete_preferences(self, registry_id, preferences_keys, json_out):
+        if not preferences_keys:
+            click.echo('Preferences keys to be removed should not be empty!', err=True)
+            sys.exit(1)
+        updated_registry = DTS.delete_preferences(registry_id, preferences_keys)
+        self._print_registry(updated_registry, json_out)
+
+    def _convert_preferences_list_to_dict(self, preferences_list):
+        preferences_dict = {}
+        for preference_entry in preferences_list:
+            preference_value_and_key = preference_entry.split(self._PREF_DELIMITER)
+            if len(preference_value_and_key) != 2:
+                click.echo('Error [%s]: preference should contain exactly one delimiter!' % preference_entry, err=True)
+                sys.exit(1)
+            else:
+                preferences_dict[preference_value_and_key[0]] = preference_value_and_key[1]
+        return preferences_dict
+
+    def _print_registry(self, registry, json_out):
+        if json_out:
+            self._print_dts_json(registry)
+        else:
+            self._print_registries_prettytable([registry])
+
+    def _print_dts_json(self, object):
+        click.echo(json.dumps(object, cls=DtsEncoder))
+
+    def _print_registries_prettytable(self, registries):
+        table = self._init_table()
+        for registry in registries:
+            table.add_row(self._convert_registry_to_prettytable_row(registry))
+        click.echo(table)
+
+    def _init_table(self):
+        table = prettytable.PrettyTable()
+        table.field_names = self._DTS_TABLE_HEADERS
+        table.align = "l"
+        table.header = True
+        return table
+
+    def _convert_registry_to_prettytable_row(self, registry):
```

Comment:
Let's show preferences only if a single registry is shown. Additionally let's use slightly different output view for single registry. As a result it would be nice to have something similar to how `pipe view-tools` behaves.

```
> pipe dts list
Id Name CreatedDate
1 dts1 date1
2 dts2 date2

> pipe dts list dts1
Id: 1
Name: dts1
Url: url1
Created date: date
Schedulable: true
Prefixes:
prefix1
prefix2
Preferences:
preferences1=value1
preferences2=value2
```